src/Passes/Source/AllocationManager/AllocationManager.hpp

```diff
@@ -0,0 +1,94 @@
+#pragma once
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "AllocationManager/IAllocationManager.hpp"
+
+#include "Llvm/Llvm.hpp"
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+namespace microsoft
+{
+namespace quantum
+{
+    /// AllocationManager is a simple qubit and results allocator that can be used at compile-time.
+    /// It is based on an assumption that all qubit allocating function calls are inlined and that
+    /// qubits/results can be allocated with strictly growing IDs.
+    class BasicAllocationManager : public IAllocationManager
+    {
+      public:
+        /// Defines a named register/memory segment with start
+        /// position, end position and size.
+        struct MemoryMapping
```

Comment:
Is it really a _mapping_? (which maps from one space to the other space)
Intuitively I feel that you could mean something like "MemoryAllocation" or "AllocatedMemoryBlock".

Reply:
It is a mapping in the sense that it holds pairs of name and memory segment, but it is not a map. That said, I agree that `AllocatedMemoryBlock` is a better name.